### PR TITLE
Add Exit system call class to libtock-c.

### PR DESCRIPTION
**Pull request by phil-levis**
This adds tock_restart and tock_exit to libtock-c, which invoke exit-restart and exit-terminate. It builds on the `yield_2.0` branch/PR #161 .

It adds two test apps, restart and terminate to test their operation. It also adds a trivial test app "fault" for when we want to test behavior on faulting.

This has been tested on cortex-m.

### CODE REVIEW

File: examples/tests/exit/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Test `restart`
+====================
+
+This tests whether a process can restart. It calls printf, waits, then
+restarts. If it runs correctly you should see the same text printed
+multiple times.
```
Comment (hudson-ayers) [inline examples/tests/exit/README.md:6]: Needs to be updated

**Review by hudson-ayers — APPROVED** [review]
Looks good to me!

**Review by alistair23 — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]
bors d+

**Pull request merged**